### PR TITLE
Study-view prototype: Make pie chart slice color consistent with selection/filters

### PR DESCRIPTION
**Pull request by kalletlak**

### CODE REVIEW

File: src/pages/studyView/charts/ChartContainer.tsx [hunk 1]
```diff
@@ -48,6 +47,7 @@ export class ChartContainer extends React.Component<IChartContainerProps, {}> {
     constructor(props: IChartContainerProps) {
         super(props);
         let fileName = props.clinicalAttribute.displayName.replace(/[ \t]/g,'_')
+        let  clinicalDataType = this.props.clinicalAttribute.patientAttribute?ClinicalDataType.PATIENT:ClinicalDataType.SAMPLE;
```
Comment (alisman) [inline src/pages/studyView/charts/ChartContainer.tsx]: we should have linting for this, but please always put spaces between operators

File: src/pages/studyView/charts/ChartContainer.tsx [hunk 2]
```diff
@@ -103,7 +103,7 @@ export class ChartContainer extends React.Component<IChartContainerProps, {}> {
     }
 
     @computed get data(){
-        return this.props.promise.result
+        return _.isUndefined(this.props.promise.result) ? [] : this.props.promise.result;
```
Comment (alisman) [inline src/pages/studyView/charts/ChartContainer.tsx]: this.props.promise.result || []

Comment (alisman) [reply src/pages/studyView/charts/ChartContainer.tsx]: also, it is possible in mobxpromise to make a default result.   not sure if you can do it here if you're using cache but worth a look.  not a big deal either way.

File: src/pages/studyView/charts/pieChart/PieChart.tsx [hunk 3]
```diff
@@ -63,6 +62,29 @@ export default class PieChart extends React.Component<IPieChartProps, {}> implem
...
+    @computed get annotatedData() {
+        return this.props.data.map(slice => {
+            let color = this.colorSet[slice.value];
+            if (_.isUndefined(color)) {
+                if (slice.value.toLowerCase().includes('na')) {
+                    color = NA_COLOR;
+                } else {
+                    color = COLORS[Object.keys(this.colorSet).length];
+                }
+                this.colorSet[slice.value] = color;
```
[the remark this answers is not in the document]
Comment (kalletlak) [reply src/pages/studyView/charts/pieChart/PieChart.tsx]: If you observe the new study-view example http://www.cbioportal.org/beta/newstudy#/study?studyId=hnsc_tcga slice colors are changed in when filters are applied. Slice colors need to be consistent all time. May be I can do the same as what I did for getting data(using MobxPromiseCache)

[the remark this answers is not in the document]
Comment (alisman) [reply src/pages/studyView/charts/pieChart/PieChart.tsx]: I don't think you should need MobxPromiseCache.   It sounds to me like colorSet is just a computation of?

[the remark this answers is not in the document]
Comment (alisman) [reply src/pages/studyView/charts/pieChart/PieChart.tsx]: unless i'm missing something, it doesn't look like there is any reference to this.colorset outside of annotatedData. if that's true, then perhaps just keep colorset as a local variable of the annotatedGetter.  Would that work?

**Pull request merged**